x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.test.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { coreMock } from '@kbn/core/public/mocks';
+import { render, screen } from '@testing-library/react';
+import { TestProvider } from '../../test/test_provider';
+import { ComplianceDashboard } from '..';
+import { useFindingsStatusApi } from '../../common/api/use_findings_status_api';
+import { NO_DATA_CONFIG_TEXT } from './translations';
+import { useCisKubernetesIntegration } from '../../common/api/use_cis_kubernetes_integration';
+import * as TEXT from './translations';
+import { useComplianceDashboardDataApi } from '../../common/api/use_compliance_dashboard_data_api';
+
+jest.mock('../../common/api/use_findings_status_api');
+jest.mock('../../common/api/use_cis_kubernetes_integration');
+jest.mock('../../common/api/use_compliance_dashboard_data_api');
+
+describe('<ComplianceDashboard />', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    (useCisKubernetesIntegration as jest.Mock).mockImplementation(() => ({
+      isSuccess: true,
+      isLoading: false,
+      data: { item: { status: 'installed' } },
+    }));
+  });
+
+  const renderComplianceDashboardPage = () => {
+    const mockCore = coreMock.createStart();
+
+    render(
+      <TestProvider
+        core={{
+          ...mockCore,
+          application: {
+            ...mockCore.application,
+            capabilities: {
+              ...mockCore.application.capabilities,
+              // This is required so that the `noDataConfig` view will show the action button
+              navLinks: { integrations: true },
+            },
+          },
+        }}
+      >
+        <ComplianceDashboard />
+      </TestProvider>
+    );
+  };
+
+  it('shows noDataConfig when findings status is inapplicable', () => {
+    (useFindingsStatusApi as jest.Mock).mockImplementation(() => ({
+      data: { status: 'inapplicable' },
+    }));
+    (useComplianceDashboardDataApi as jest.Mock).mockImplementation(() => ({
+      data: undefined,
+    }));
+
+    renderComplianceDashboardPage();
+
+    Object.values(NO_DATA_CONFIG_TEXT).forEach((text) =>
+      expect(screen.getAllByText(text)[0]).toBeInTheDocument()
+    );
+    expect(screen.queryByText(TEXT.CLOUD_POSTURE_SCORE)).not.toBeInTheDocument();
+  });
+
+  it('shows dashboard when findings status is applicable', () => {
+    (useFindingsStatusApi as jest.Mock).mockImplementation(() => ({
+      isLoading: false,
+      isSuccess: true,
+      data: { status: 'applicable' },
+    }));
+
+    (useComplianceDashboardDataApi as jest.Mock).mockImplementation(() => ({
+      isSuccess: true,
+      isLoading: false,
+      data: {
```

Comment:
Consider extracting the data out of the test body and into a constant for easier reading of the test flow